### PR TITLE
Remove keys from HashSet when items expire

### PR DESCRIPTION
**Pull request by rajmondburgaj**
Removing keys from internal hashset if the item already expires from the IMemoryCache

### CODE REVIEW

File: Src/Coravel/Cache/InMemoryCache.cs [hunk 1]
```diff
@@ -72,12 +69,22 @@ public void Forget(string key)
...
             }
 
             this._keys.Clear();
         }
+
+        private void EvictionCallback(object key, object value, EvictionReason reason, object state)
+        {
+            if (reason == EvictionReason.Expired || reason == EvictionReason.Capacity || reason == EvictionReason.Removed)
+            {
+                _keys.Remove(key.ToString());
```
Comment (jamesmh) [inline Src/Coravel/Cache/InMemoryCache.cs]: Just for consistency could you add `this._keys....` ?

File: Src/Coravel/Cache/InMemoryCache.cs [hunk 2]
```diff
@@ -38,10 +34,11 @@ public async Task<TResult> RememberAsync<TResult>(string key, Func<Task<TResult>
             this._keys.Add(key);
 
             return await this._cache.GetOrCreateAsync(key, async entry =>
-           {
-               entry.AbsoluteExpiration = DateTimeOffset.Now.Add(expiresIn);
-               return await cacheFunc();
-           });
+            {
+                entry.AbsoluteExpiration = DateTimeOffset.Now.Add(expiresIn);
+                entry.RegisterPostEvictionCallback(EvictionCallback);
```
Comment (jamesmh) [inline Src/Coravel/Cache/InMemoryCache.cs]: For consistency add `this` please?

File: Src/Coravel/Cache/InMemoryCache.cs [hunk 3]
```diff
@@ -29,6 +24,7 @@ public TResult Remember<TResult>(string key, Func<TResult> cacheFunc, TimeSpan e
             return this._cache.GetOrCreate(key, entry =>
             {
                 entry.AbsoluteExpiration = DateTimeOffset.Now.Add(expiresIn);
+                entry.RegisterPostEvictionCallback(EvictionCallback);
```
Comment (jamesmh) [inline Src/Coravel/Cache/InMemoryCache.cs]: For consistency add `this` please?

File: Src/Coravel/Cache/InMemoryCache.cs (hunk 1 above)
Comment (rajmondburgaj) [reply Src/Coravel/Cache/InMemoryCache.cs]: Made the requested changes, please let me know if there is anything else to be changed :)

**Pull request merged**